core/src/worker/api.ts

```diff
@@ -36,6 +36,22 @@ export async function fetchDataFeed({
   }
 }
 
+export async function fetchDataFeedByAggregatorId({
+  aggregatorId,
+  logger
+}: {
+  aggregatorId: bigint
+  logger: Logger
+}): Promise<{ timestamp: string; value: bigint }> {
```

Comment:
I think we could reuse this type

```suggestion
}): Promise<IAggregate> {
```

Reply:
IAggregate contains `id`, and `aggregatorId` which is not referenced from client codes. if using `IAggregate`, I'd probably have to modify api to include those variables

Reply:
new dataType has been created

```
export interface IAggregateById {
  timestamp: string
  value: bigint
}
```